packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp

```diff
@@ -1298,15 +1298,8 @@ class Impl {
 #endif
       TEUCHOS_ASSERT(sweep >= 1);
       if ( ! force && (sweep - 1) % sweep_step_) return false;
-      if (collective_) {
-#ifdef HAVE_MPI
-# if MPI_VERSION >= 3
-        MPI_Wait(&mpi_request_, MPI_STATUS_IGNORE);
-# else
-        // Do nothing.
-# endif
-#endif
-      }
+      if (collective_)
+        tpetra_comm_request_->wait();
```

Comment:
If `tpetra_comm_request_` never leaves the scope of this method without having wait() called on it, then you don't have to keep it around as an instance field.  There's no performance benefit to that as far as I can tell.

Reply:
@mhoemmen, the request is made in ireduce(), and the wait occurs in check_done(). So I think I need the instance variable.